src/NotificationTemplate.php

```diff
@@ -259,6 +259,17 @@ public function getTemplateByLanguage(
         $tid  = $language;
         $tid .= serialize($additionnaloption);
 
+        // Include user-specific data in cache key to ensure each user gets unique data
+        // This is crucial for plugins that generate user-specific tokens or data
+        if (isset($user_infos['users_id'])) {
+            $tid .= '_user_' . $user_infos['users_id'];
+        } elseif (isset($user_infos['email'])) {
+            $tid .= '_email_' . $user_infos['email'];
+        } else {
+            // Fallback: use a unique identifier to avoid cache collision
+            $tid .= '_unique_' . uniqid();
```

Comment:
If you always fallback to a unique cache identifier, as long as a same user can only receive one notification for a given event, then it means that you completely disable the email templates cache.
Is this really needed ?